oracle/datadog_checks/oracle/oracle.py

```diff
@@ -165,44 +142,38 @@ def check(self, _):
 
     @property
     def _connection(self):
-        """Creates a connection or raises an exception"""
         if self._cached_connection is None:
-            if self.can_use_oracle_client():
-                self._cached_connection = self._oracle_client_connect()
-            elif JDBC_IMPORT_ERROR:
-                self._connection_errors += 1
+            if self.can_use_jdbc():
+                try:
+                    self._cached_connection = self._jdbc_connect()
+                except Exception as e:
+                    self.log.error("The JDBC connection failed with the following error: %s", str(e))
+                    self._connection_errors += 1
+            else:
+                self._cached_connection = self._oracle_connect()
+        return self._cached_connection
+
+    def can_use_jdbc(self):
+        if self._jdbc_driver:
```

Comment:
Should there be an `else` for this `if` statement? I don't see what would happen if `self._jdbc_driver` is false

Reply:
You're absolutely right, thanks for the review